docs/proposal/AI-gateway.md

```diff
@@ -0,0 +1,293 @@
+
+2. `External Processing Filter` will process header/body/trailer synchronously, while AI traffic is mainly processed based on body. So the processing result of header has been returned to Envoy at this time. Fortunately, from the test results, the processed header will not be sent to upstream immediately, but cached, so it will not cause trouble to the backend model.
+
+3. After the traffic is hijacked by `External Processing Filter`, the general traffic management functions configured on the gateway (i.e., connection-level and request-level management) are actually skipped. But after careful consideration, you will find that this is not a problem, because the traffic accessing the model will become east-west traffic, which is the inherent capabilities of Kmesh. We can still perform general traffic management functions on the traffic sent from the AI Plugin to the model.
+
+#### AI Plugin CRD
+
+#### VirtualModel
+
+`VirtualModel` defines the traffic management strategy from the model specified by the user in the request (which can even be set to empty) to the model actually deployed on the backend.
+
+API:
+
+```go
+// VirtualModel is the Schema for the virtualmodels API.
+type VirtualModel struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   VirtualModelSpec   `json:"spec,omitempty"`
+	Status VirtualModelStatus `json:"status,omitempty"`
+}
+
+// VirtualModelSpec defines the desired state of VirtualModel.
+type VirtualModelSpec struct {
+	Models []string `json:"models,omitempty"`
+
+	Rules []*Rule `json:"rules"`
+}
+
+type Rule struct {
+	Name  string              `json:"name,omitempty"`
+	Match []*MatchCondition   `json:"match,omitempty"`
+	Route []*RouteDestination `json:"route"`
+
+	Timeout *metav1.Duration `json:"timeout,omitempty"`
+	Retries *Retry           `json:"retries,omitempty"`
+}
+
+type MatchCondition struct {
+	// Implement if necessary.
+}
+
+type RouteDestination struct {
+	Destination *Destination `json:"destination"`
+	Weight      int32        `json:"weight,omitempty"`
+}
+
+type Destination struct {
+	Host  string `json:"host"`
+	Model string `json:"model"`
+}
+
+type Retry struct {
+	Attempts int32 `json:"attempts"`
+}
+```
+
+The config example of the above Go code API converted into yaml format is as follows:
+
+```yaml
+apiVersion: ai.kmesh.net/v1alpha1
+kind: VirtualModel
+metadata:
+  name: llama
+  namespace: ollama
+spec:
+  models:
+  - "llama"
+  rules:
+  - name: "llama-route-32"
+  # match: 
+    route:
+    - destination:
+        host: "http://ollama.ollama.svc.cluster.local:11434"

```

Comment:
As we discussed, we need to explicitly express service fqdn and port scheme. 

Further more if this model provider use a different url, how can we specify it